pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -254,20 +254,49 @@ public SuccessResponse deleteTable(
 
     List<String> tablesDeleted = new LinkedList<>();
     try {
-      if (tableType != TableType.REALTIME && !TableNameBuilder.REALTIME.tableHasTypeSuffix(tableName)) {
+      boolean tableExist = false;
+      if (verifyTableType(tableName, tableType, tableType.OFFLINE)) {
+        tableExist = _pinotHelixResourceManager.hasOfflineTable(tableName);
+        // Even the table name does not exist, still go on to delete remaining table metadata in case a previous delete
+        // did not complete.
         _pinotHelixResourceManager.deleteOfflineTable(tableName);
-        tablesDeleted.add(TableNameBuilder.OFFLINE.tableNameWithType(tableName));
+        if (tableExist) {
+          tablesDeleted.add(TableNameBuilder.OFFLINE.tableNameWithType(tableName));
+        }
       }
-      if (tableType != TableType.OFFLINE && !TableNameBuilder.OFFLINE.tableHasTypeSuffix(tableName)) {
+      if (verifyTableType(tableName, tableType, tableType.REALTIME)) {
+        tableExist = _pinotHelixResourceManager.hasRealtimeTable(tableName);
+        // Even the table name does not exist, still go on to delete remaining table metadata in case a previous delete
+        // did not complete.
         _pinotHelixResourceManager.deleteRealtimeTable(tableName);
-        tablesDeleted.add(TableNameBuilder.REALTIME.tableNameWithType(tableName));
+        if (tableExist) {
+          tablesDeleted.add(TableNameBuilder.REALTIME.tableNameWithType(tableName));
+        }
+      }
+      if (!tableExist) {
+        throw new ControllerApplicationException(LOGGER,
```

Comment:
This exception will be caught by line 282. You need to move this outside of the try-catch block.
You can check whether `tablesDeleted.isEmpty()`

Reply:
Good point. Done. @Jackie-Jiang